src/Microsoft.Data.SqlClient/tests/ManualTests/SQL/DataReaderTest/DataReaderCancellationTest.cs

```diff
@@ -0,0 +1,49 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Data.SqlClient.ManualTesting.Tests
+{
+    public class DataReaderCancellationTest
+    {
+        /// <summary>
+        /// Test ensures cancellation token is registered before ReadAsync starts processing results from TDS Stream,
+        /// such that when Cancel is triggered, the token is capable of canceling reading further results.
+        /// </summary>
+        /// <returns>Async Task</returns>
+        [ConditionalFact(typeof(DataTestUtility), nameof(DataTestUtility.AreConnStringsSetup))]
+        public static async Task CancellationTokenIsRespected_ReadAsync()
+        {
+            const string longRunningQuery = @"
+with TenRows as (select Value from (values (1), (2), (3), (4), (5), (6), (7), (8), (9), (10)) as TenRows (Value)),
+    ThousandRows as (select A.Value as A, B.Value as B, C.Value as C from TenRows as A, TenRows as B, TenRows as C)
+select *
+from ThousandRows as A, ThousandRows as B, ThousandRows as C;";
```

Comment:
Reading a billion rows could take a long time. Would this be a concern in the test suite, on the off chance that this test is regressed?

Reply:
So it shouldn't regress :)
But I'll wait for the CI builds to say that - I don't think that should be a problem IMO.

Reply:
The danger is that someone works in this area and does regress it within their branch, and every CI job takes 60 minutes to run and times out before they figure out what's going on. Just an inconvenience.

Reply:
I don't think it should be that bad as it takes around 60 seconds to complete query execution locally. We cannot add a 'WAITFOR DELAY' or any procedure call, as that would cause delay in ExecuteReaderAsync and not ReadAsync. And ReadAsync can only be delayed with a large dataset. If you have a better query, let me know and I can evaluate. We also want to be able to consistently rely in tests, as intermittent test failures are more painful to identify regressions.

Reply:
> @JRahnama any clue why CI builds are not getting triggered?

Can you commit some changes? Azure had some difficulties resolving Default branch for manual and scheduled builds address which is now solved and should be working.

Reply:
@JRahnama - I've updated the service connection so it seems to be triggering CI builds now. Just FYI, in case you wonder later 🙂